epigraphhub/models/ngboost_models.py

```diff
@@ -0,0 +1,740 @@
+                # print(df_lag.index,fit_target.index)
+
+        models = []
+
+        X_train = df_lag.iloc[:-1]
+
+        for T in range(1, horizon_forecast + 1):
+            # training of the model with all the data available
+
+            # print(T)
+
+            tgt = targets[T][: len(X_train)]
+
+            i = 0
+
+            while i < len(tgt):
+                if tgt[i] <= 0:
+
+                    tgt[i] = 0.01
+                i = i + 1
+
+            # if not os.path.exists(f'../opt/models/saved_models/ml/ngboost_{target_name}_{T}D.joblib'):
+
+            model = NGBRegressor(
+                Base=default_tree_learner,
+                Dist=LogNormal,
+                Score=LogScore,
+                natural_gradient=True,
+                verbose=False,
+                col_sample=0.9,
+                n_estimators=300,
+                learning_rate=0.1,
+                validation_fraction=0.25,
+                early_stopping_rounds=50,
+            )
+
+            model.fit(X_train.iloc[: len(tgt)], tgt)
+
+            dump(
+                model,
+                f"../opt/models/saved_models/ml/ngboost_{target_name}_{T}D.joblib",
+            )
+
+            models.append(model)
+
+    return models
+
+
+def rolling_forecast(target_name, data, ini_date, horizon_forecast=14, maxlag=14):
+
+    # print(data.index[-1])
+    data = data.iloc[:-3]
+
+    # print(data.index[-1])
+    target = data[target_name]
+
+    df_lag = build_lagged_features(copy.deepcopy(data), maxlag=maxlag)
+
+    ini_date = max(
+        df_lag.index[0], target.index[0], datetime.strptime(ini_date, "%Y-%m-%d")
+    )
+
+    df_lag = df_lag[ini_date:]
+    target = target[ini_date:]
+    target = target.dropna()
+    df_lag = df_lag.dropna()
+
+    # condition to only apply the model for datas with values different of 0
+    if np.sum(target) > 0.0:
+
+        # remove the target column and columns related with the day that we want to predict
+        df_lag = df_lag.drop(data.columns, axis=1)
+
+        # targets
+        targets = {}
+
+        for T in np.arange(1, horizon_forecast + 1, 1):
+            if T == 1:
+                targets[T] = target.shift(-(T - 1))
+            else:
+                targets[T] = target.shift(-(T - 1))[: -(T - 1)]
+        #         print(T, len(df_lag), len(fit_target))
+        #         print(df_lag.index,fit_target.index)
+
+        # forecast
+        forecasts5 = []
+        forecasts50 = []
+        forecasts95 = []
+
+        X_train = df_lag.iloc[:-1]
+
+        for T in range(1, horizon_forecast + 1):
+            # training of the model with all the data available
+
+            # print(T)
+
+            tgt = targets[T][: len(X_train)]
+
+            i = 0
+
+            while i < len(tgt):
+                if tgt[i] <= 0:
+
+                    tgt[i] = 0.01
+                i = i + 1
+
+            model = load(
+                f"../opt/models/saved_models/ml/ngboost_{target_name}_{T}D.joblib"
```

Comment:
probably you would prefer to use a variable for the root path or get the path from a environment variable